Booking check out

jaz998 opened the pull request:
The changes are primarily about the bookingcheckout.php page. 

This page takes the booking start and end time from booking2.php page and validate again whether the car is already booked before inserting the booking into the database.

Currently the info in the credit card details form on this page is not inserted into the database - this will happen in future commits. 

The codes also do not validate whether the user is logined or send confirmation email - will be done in future commits.

jaz998 committed:
Update 3c-reserve-range.js

include status==2, in which the user has logined and thus is redirected to logined page to login

jaz998 committed:
Update 2c-ajax-reserve.php

creates json response when the user is not logined

jaz998 committed:
Update 3c-reserve-range.js

user will check out (enter credit card details etc) before finalising booking

jaz998 committed:
Update 2b-lib-res.php

hard code a return value to redirect user to bookingcheckout.php.

I am stuck as the codes should return true (all okay), false (car already booked for that period), or 2 (user not logined yet and should redirect user to login page). It works fine if I hard code a value but does not work if I include the if statement below, any idea why?

//    session_start();
//    if ( !isset($_SESSION["carbuddy"]) || (isset($_SESSION["carbuddy"]) && !$_SESSION["carbuddy"]->isLoggedIn())) {
//        return 2;
//    }

jaz998 committed:
Create checkoutstyles.css

styling for checkout / payment details form

jaz998 committed:
Update 2b-lib-res.php

created bookRangeValidate function used to validate bookings on booking2.php, and bookRange function used to make bookings on bookingcheckout.php

jaz998 committed:
Update 2c-ajax-reserve.php

added validation whether the user has logined

jaz998 committed:
Create bookingcheckout.php

booking checkout page with credit card details, limited functionalities now - only makes a booking but disgard and credit card details

jaz998 committed:
Update bookingcheckout.php

added displaying no. of hours and fee per hour and total fees

jaz998 committed:
Create Payment.php

Class for making a payment

jaz998 committed:
Update Payment.php

rewrote function calFee to separate the getting/calculating FeePerHours and Hours of the booking into two seperate functions calHours and getFeePerHour

jaz998 committed:
Update bookingcheckout.php

originally calFee returns an array with (FeePerHour, Hours, TotalFee), now these are handled by individual functions calHours, getFeePerHour and calFee, rewrote the codes on this booking checkout/credit card details page to reflect that.

jaz998 committed:
writting more var to booking table

add three more variables ($FeePerHour, $Hours, $TotalFee) to the booking record of the BookingsCurrent table

jaz998 committed:
redirecting to thank-you page after booking is successful

jaz998 committed:
booking confirmation page

deleted 3d-thank-you.html and added 3d-thank-you.php to add existing styling.

jaz998 committed:
Create car-alreay-booked.php

if on booking2.php, the validation shows car already booked for period it will be redirected to this page.

jaz998 committed:
Update bookingcheckout.php

chnage redirection to 3d-thank-you.php instead of 3d-thank-you.html, just for including the page stylings.

jaz998 committed:
Update bookingcheckout.php

s3666874 commented on classes/Payment.php:
```diff
@@ -0,0 +1,99 @@
+    $this->_db = DB::getInstance();
+    $this->_sendMail = Email::getInstance();
+
+    $this->_sessionName = Config::get('sessions/session_name');
```
@jaz998 the session name is car_buddy. Is that what you want to use? Or have you changed it in another PR?

s3666874 commented on classes/Payment.php:
```diff
@@ -0,0 +1,99 @@
+    $this->PostCode = $PostCode;
+    
+
+    Session::put($this->_sessionName, $this);
```
I've had issues with putting the entire object in session. You need to serialize and deserialize it in order to store it properly.

s3666874 commented on lib/2b-lib-res.php:
```diff
@@ -106,13 +106,53 @@ function bookRange ($name, $email, $tel, $start, $end, $notes="", $vehicle_id) {
+//    if ( !isset($_SESSION["carbuddy"]) || (isset($_SESSION["carbuddy"]) && !$_SESSION["carbuddy"]->isLoggedIn())) {
+//        return 2;
+//    }
+    if(!session_start()){
```
I don't think you need to do this. Just calling session_start, PHP will start or resume an existing session automatically for you.

jaz998 commented on classes/Payment.php:
```diff
@@ -0,0 +1,99 @@
+          $ExpYear,
+          $CardCVV,
+          $Address,
+          $State,
```
testing comment

jaz998 committed:
Update Payment.php

Deleting session variable handling that I copied from User class. On a second thought don't think there is a need to use session varaibles and also I see now on master branch sessoin handling has changed.

jaz998 committed:
Update 2b-lib-res.php

!session_start() is not the right way to do it, the only reason I do this check is on my pc it tends to give a notice/warning saying session already if there is an existing session, and if I don't do this check

jaz998 commented:
@s3666874 , thanks for your comments, I've made changes accordingly, pls have a look, thanks.
